### PR TITLE
issue: 784234 fixed epoll is not cleared from vma_stats

### PR DESCRIPTION
**Pull request by DanielLibenson**
Signed-off-by: Daniel Libenson <daneilli@mellanox.com>

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-redmine-issue/22/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/139/ for details (Mellanox internal link).

**Comment by rosenbaumalex** [comment]
+1

**Comment by rosenbaumalex** [comment]
let's wait with the merge until after VMA-8.0 is formally released

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-redmine-issue/23/ for details (Mellanox internal link).

File: src/stats/stats_publisher.cpp [hunk 1]
```diff
@@ -624,8 +624,18 @@ void vma_stats_instance_create_epoll_block(int fd, iomux_func_stats_t* local_sta
...
+		for (int i=0; i<NUM_OF_SUPPORTED_EPFDS; i++) {
+			if (&g_sh_mem->iomux.epoll[i].stats == ep_func_stats) {
+				g_sh_mem->iomux.epoll[i].enabled = false;
+				g_lock_ep_stats.unlock();
+				return;
+			}
+		}
+	}
+
+	vlog_printf(VLOG_WARNING, "Can't remove epoll stat memory block\n");
```
Comment (OphirMunk) [inline src/stats/stats_publisher.cpp]: Need to follow the same pattern for removal of other components in the file. For example see 
void vma_stats_instance_remove_cq_block(cq_stats_t* local_stats_addr)

File: src/stats/stats_publisher.cpp [hunk 2]
```diff
@@ -625,17 +625,23 @@ void vma_stats_instance_remove_epoll_block(iomux_func_stats_t* local_stats_addr)
...
+			g_lock_ep_stats.unlock();
+			return;
 		}
 	}
 
-	vlog_printf(VLOG_WARNING, "Can't remove epoll stat memory block\n");
+	vlog_printf(VLOG_ERROR, "%s:%d: Could not find user pointer (%p)", __func__, __LINE__, ep_func_stats);
 	g_lock_ep_stats.unlock();
+	return;
 }
```
Comment (OphirMunk) [inline src/stats/stats_publisher.cpp]: +1

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/140/ for details (Mellanox internal link).

**Pull request merged**